test/cache_test.rb

```diff
@@ -1,6 +1,12 @@
 require 'test_helper'
 require 'tmpdir'
 require 'tempfile'
+
+InheritedRoleSerializer = Class.new(RoleSerializer) do
+  cache key: 'inherited_role', only: [:name, :special_attribute]
+  attribute :special_attribute
+end
```

Comment:
Better to include this in the `class CacheTest` so it's not in the global scope, but not a blocker